src/tests/prepare.spec.js

```diff
@@ -65,6 +64,80 @@ describe('prepare', () => {
     await prepare(<MessageBox />);
   });
 
+  it('Should throw exception', async () => {
+    const doAsyncSideEffect = sinon.spy(async () => {
+      throw new Error('Err');
+    });
+    const prepareUsingProps = async ({ text }) => {
+      await doAsyncSideEffect(text);
+    };
+    const App = prepared(prepareUsingProps)(({ text }) => <div>{text}</div>);
+    try {
+      await prepare(
+        <App text="foo">
+          <App text="foo" />
+          <App text="foo" />
+          <App text="foo" />
+          <App text="foo" />
+        </App>
+      );
+    } catch (err) {
+      t.assert(doAsyncSideEffect.calledOnce, 'Should be called once times');
+      return;
+    }
+    t.assert(false, 'It should throw');
```

Comment:
Didn't find a way (quick) to to a `shoud throw` with the current testing lib, but this "works".